apps/api/src/app/workflows-v2/usecases/preview/preview.usecase.ts

```diff
@@ -114,11 +119,18 @@ export class PreviewUsecase {
         };
       }
 
-      const mergedVariablesExample = this.mergeVariablesExample(workflow, previewTemplateData, commandVariablesExample);
+      const previewPayloadExample = this.mergePayloadExample(
+        workflow,
+        isEnhancedDigestEnabled
+          ? enhanceEventCountValue(previewTemplateData.payloadExample)
```

Comment:
The `enhanceEventCountValue` assigns the `events.length` value to the `eventCount` variable. Before we have the logic that "duplicates" x3 times the `event` object in the `events` array. That's why the enhancement is here at the place where we construct the example payload that is later sent to the bridge.